### ISSUE TITLE
NON FINITE DEPTH FOUND

### ISSUE THREAD

**Issue by yehaifengcaiji** [FVCOM-GitHub/fvcom50#23]
Hi, when I run the fvcom, and report that:

NON FINITE DEPTH FOUND
FVCOM MODEL HAS BECOME UNSTABLE

Could you please help me?

Thanks,
Haifeng

**Comment by snaled** [comment]
I'm having the same issue. Pls did you solve it

**Comment by liesvyvall** [comment]
Hi, @yehaifengcaiji @snaled  Are you using ssh assimilation? I Thing this is related to having nans in the input file, check the mod_utils.F file:

```
#  if !defined (GFORTRAN) && !defined (ABSOFT)
    IF(ISNAN(RBUF))THEN 
       IF (PRESENT(MSG)) THEN
          CALL FATAL_ERROR("SHUTDOWN_CHECK FOUND NON FINITE VALUE:",&
               & MSG )
       ELSE
          CALL FATAL_ERROR('NON FINITE VALUE (DEPTH?) FOUND',&
               & 'MODEL HAS BECOME UNSTABLE')
       END IF

    END IF
#  else
```

**Comment by victorbka** [comment]
I'm having the same issue，May I ask how you resolved this issue

**Comment by yehaifengcaiji** [comment]
Yes, I solved it by setting a sponge layer and change the time step.

**Comment by yehaifengcaiji** [comment]
> I'm having the same issue，May I ask how you resolved this issue


Ok, I solved it by setting a sponge layer and changing the time step.

**Comment by victorbka** [comment]
> > I'm having the same issue，May I ask how you resolved this issue我遇到了同样的问题，请问您是怎么解决这个问题的
> 
> Ok, I solved it by setting a sponge layer and changing the time step.好的，我通过设置海绵层和更改时间步长解决了这个问题。

Oh，thanks you so much
